source/MaterialXGenMdl/Nodes/CompoundNodeMdl.cpp

```diff
@@ -135,18 +182,32 @@ void CompoundNodeMdl::emitFunctionSignature(const ShaderNode&, GenContext& conte
 
     if (!_returnStruct.empty())
     {
-        // Define the output struct.
-        shadergen.emitLine("struct " + _returnStruct, stage, false);
-        shadergen.emitScopeBegin(stage, Syntax::CURLY_BRACKETS);
-        for (const ShaderGraphOutputSocket* output : _rootGraph->getOutputSockets())
+        if (_unrollReturnStructMembers)
         {
-            shadergen.emitLine(syntax.getTypeName(output->getType()) + " mxp_" + output->getName(), stage);
+            const auto fieldName = context.getUserData<GenUserDataString>("returnStructFieldName");
+
+            // Begin function signature.
+            const ShaderGraphOutputSocket* outputSocket = _rootGraph->getOutputSocket(fieldName->getValue());
```

Comment:
Do we need a nullptr check on outputSocket before the dereference below? Could throw an error message instead of crashing, if there is a chance of the fieldName not being found.

Reply:
made a suggestion (in the next commit)